src/bmp/BMPParser.cc

```diff
@@ -214,14 +240,18 @@ void Parser::parse(byte *buf, int bufSize, byte *format){
   // Check if pre-multiplied alpha is used
   bool premul = format ? format[4] : 0;
 
-  // Bitmap data starts at the lower left corner
-  for(int y = h - 1; y != -1; y--){
+  // Don't perform overrun checks in the loop
+  #undef GET_METHOD
+  #define GET_METHOD getc

```

Comment:
In the category of macros-are-evil... I think this would be better done as a template parameter, something like:

```cpp
// BMPParser.h
template <typename T, bool skipcheck = false> T get();

// BMPParser.cc
template <typename T, bool skipcheck> T get() {
  if (!skipcheck) CHECK_OVERRUN(sizeof(T), T);
  // ...
}
```

Then however you prefer ... new `#define`s for like `U1UC` ("U1 unchecked"), or `get<uint8_t, true>()`,